### PR TITLE
Automation tests for Posts

### PR DESCRIPTION
**Pull request by deleugpn**
- Test store method;
- Test CreatePostRequest;
- Test destroy method;

As pointed out in #54, the `store` test is not passing because of a bug.

### CODE REVIEW

**Comment by NowNewStart** [comment]
I'd use the user factory and would do `actingAs($user)`

**Comment by deleugpn** [comment]
@NowNewStart that's my first choice as well, but that still need to be accounted for in the PostController.

File: tests/Integration/PostTest.php [hunk 1]
```diff
@@ -5,20 +5,22 @@
...
+        $user = factory(User::class)->create();
 
-        // @TODO: are we going to provide the user_id as parameter or authentication?
-        $response = $this->post(route('forum.posts.store'), [
-            'discussion' => $discussion->id,
-            'content' => 'blah'
-        ]);
+        $this->actingAs($user)
+            ->post(route('forum.posts.store'), [
+                'discussion' => $discussion->id,
```
Comment (AndreasElia) [inline tests/Integration/PostTest.php]: Pretty sure this needs to be `'discussion_id' => ...`

File: tests/Integration/PostTest.php [hunk 2]
```diff
@@ -16,7 +16,7 @@ public function test_create_post()
 
         $this->actingAs($user)
             ->post(route('forum.posts.store'), [
-                'discussion' => $discussion->id,
+                'discussion_id' => $discussion->id,
                 'content' => 'blah'
```
Comment (AndreasElia) [inline tests/Integration/PostTest.php]: Please use faker for any content.

File: database/factories/ModelFactory.php [hunk 3]
```diff
@@ -3,6 +3,7 @@
 use AndreasElia\Forum\Models\Discussion;
 use AndreasElia\Forum\Models\Group;
 use AndreasElia\Forum\Models\Post;
+use AndreasElia\Tests\stubs\models\User;
```
Comment (Dylan-DPC) [inline database/factories/ModelFactory.php]: stubs and models folder should be in Title Case

File: tests/Integration/PostTest.php [hunk 4]
```diff
@@ -2,15 +2,45 @@
...
 
+use AndreasElia\Forum\Models\Discussion;
+use AndreasElia\Forum\Models\Post;
 use AndreasElia\Tests\IntegrationTestCase;
+use AndreasElia\Tests\stubs\models\User;
 
 class PostTest extends IntegrationTestCase
 {
-    public function testCreatePost()
+    public function test_create_post()
```
Comment (Dylan-DPC) [inline tests/Integration/PostTest.php]: i prefer camelCase instead. Let's keep it consistent with it

File: tests/Integration/PostTest.php [hunk 5]
```diff
@@ -2,15 +2,45 @@
...
+            ->post(route('forum.posts.store'), [
+                'discussion' => $discussion->id,
+                'content' => 'blah'
+            ])
+            ->assertResponseStatus(201);
 
         $this->seeInDatabase('posts', ['content' => 'blah']);
     }
+
+    public function test_create_post_validation()
```
Comment (Dylan-DPC) [inline tests/Integration/PostTest.php]: camelCase

File: tests/Integration/PostTest.php [hunk 6]
```diff
@@ -2,15 +2,45 @@
...
+    {
+        $this->post(route('forum.posts.store'))
+            ->assertResponseStatus(302)
+            ->assertValidationContains('content.required', __('The content is required.'));
+    }
+
+    /**
+     * @group f
+     */
+    public function test_destroy_post()
```
Comment (Dylan-DPC) [inline tests/Integration/PostTest.php]: camelCase

File: tests/stubs/models/User.php [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace AndreasElia\Tests\stubs\models\User;
+namespace AndreasElia\Tests\stubs\models;
```
Comment (Dylan-DPC) [inline tests/stubs/models/User.php]: namespace should be in Title Case

File: src/Requests/Posts/CreatePostRequest.php [hunk 8]
```diff
@@ -27,4 +27,11 @@ public function rules()
             'content' => 'required',
         ];
     }
+
+    public function messages()
+    {
+        return [
+            'content.required' => __('The content is required.')
```
Comment (AndreasElia) [inline src/Requests/Posts/CreatePostRequest.php]: What is this for? The default request message should be fine for this field, no?

**Pull request merged**